oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/test/java/org/apache/skywalking/oap/server/receiver/envoy/als/K8sHTTPAnalysisTest.java

```diff
@@ -0,0 +1,164 @@
+            Assert.assertEquals("", incoming.getSourceServiceName());
+            Assert.assertEquals("productpage", incoming.getDestServiceName());
+            Assert.assertEquals(DetectPoint.server, incoming.getDetectPoint());
+        }
+    }
+
+    @Test
+    public void testSidecar2SidecarServerMetric() throws IOException {
+        MockK8sAnalysis analysis = new MockK8sAnalysis();
+        try (InputStreamReader isr = new InputStreamReader(getResourceAsStream("envoy-mesh-server-sidecar.msg"))) {
+            StreamAccessLogsMessage.Builder requestBuilder = StreamAccessLogsMessage.newBuilder();
+            JsonFormat.parser().merge(isr, requestBuilder);
+
+            analysis.analysis(requestBuilder.getIdentifier(), requestBuilder.getHttpLogs().getLogEntry(0), Role.SIDECAR);
+
+            Assert.assertEquals(1, analysis.metrics.size());
+
+            ServiceMeshMetric incoming = analysis.metrics.get(0);
+            Assert.assertEquals("productpage", incoming.getSourceServiceName());
+            Assert.assertEquals("review", incoming.getDestServiceName());
+            Assert.assertEquals(DetectPoint.server, incoming.getDetectPoint());
+        }
+    }
+
+    @Test
+    public void testSidecar2SidecarClientMetric() throws IOException {
+        MockK8sAnalysis analysis = new MockK8sAnalysis();
+        try (InputStreamReader isr = new InputStreamReader(getResourceAsStream("envoy-mesh-client-sidecar.msg"))) {
+            StreamAccessLogsMessage.Builder requestBuilder = StreamAccessLogsMessage.newBuilder();
+            JsonFormat.parser().merge(isr, requestBuilder);
+
+            analysis.analysis(requestBuilder.getIdentifier(), requestBuilder.getHttpLogs().getLogEntry(0), Role.SIDECAR);
+
+            Assert.assertEquals(1, analysis.metrics.size());
+
+            ServiceMeshMetric incoming = analysis.metrics.get(0);
+            Assert.assertEquals("productpage", incoming.getSourceServiceName());
+            Assert.assertEquals("detail", incoming.getDestServiceName());
+            Assert.assertEquals(DetectPoint.client, incoming.getDetectPoint());
+        }
+    }
+
+    public static class MockK8sAnalysis extends K8sALSServiceMeshHTTPAnalysis {
+        private List<ServiceMeshMetric> metrics = new ArrayList<>();
+
+        @Override
+        protected void forward(ServiceMeshMetric metric) {
+            metrics.add(metric);
+        }
+
+        @Override
+        protected ServiceMetaInfo find(String ip, int port) {
+            switch (ip) {
+                case "10.44.2.56":
+                    return new ServiceMetaInfo("ingress", "ingress-Inst");
+                case "10.44.2.54":
+                    return new ServiceMetaInfo("productpage", "productpage-Inst");
+                case "10.44.6.66":
+                    return new ServiceMetaInfo("detail", "detail-Inst");
+                case "10.44.2.55":
+                    return new ServiceMetaInfo("review", "detail-Inst");
```

Comment:
These IPs should be all pod IPs, @hanahmily please verify them mapping to the right service names.